mtp_api/apps/core/management/commands/dump_and_upload_for_ap.py

```diff
@@ -0,0 +1,53 @@
+import textwrap
+import datetime
+from tempfile import NamedTemporaryFile
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    """
+    Dump and upload data to S3 for Analytical Platform
+    """
+    help = textwrap.dedent(__doc__).strip()
+
+    def add_arguments(self, parser):
+        parser.add_argument('bucket_name', help='AWS S3 global bucket name')
+
+    def handle(self, *args, **options):
+        file_path_credits = credits_temp_file_generation()
+        file_path_disbursements = disbursements_temp_file_generation()
+
+        bucket_name = options['bucket_name']
+
+        object_name = todays_date()
+
+        dump_for_ap_command(file_path_credits, file_path_disbursements)
+        upload_dump_for_ap_command(file_path_credits, file_path_disbursements, bucket_name, object_name)
+
+
+def dump_for_ap_command(file_path_credits, file_path_disbursements):
+    call_command('dump_for_ap', 'disbursements', file_path_credits, after=yesterdays_date(), before=todays_date())
+    call_command('dump_for_ap', 'credits', file_path_disbursements, after=yesterdays_date(), before=todays_date())
```

Comment:
the date could change between these two lines, it would be safer to only calculate the ranges once
(although actually, this point is irrelevant if you have the after/before as arguments from the point above)

Reply:
the date could still change between these lines

Reply:
oh and you have disbursements and credits the wrong way round